esp/templates/program/modules/resourcemodule/classrooms.html

```diff
@@ -74,3 +97,71 @@ <h2>Step 3: Classrooms</h2>
 {% endifequal %}
 
 <br />
+
+<!--This branch has a fix for when the form is originally hidden-->
+<script src="https://rawgit.com/elo80ka/django-dynamic-formset/f5f68c818953ddf43ffa7cb2e9aa391c3d4dfb47/src/jquery.formset.js"></script>
+<!--jQuery.initialize plugin is created to help maintain dynamically created elements on the page-->
+<script src="https://rawgit.com/pie6k/jquery.initialize/master/jquery.initialize.js"></script>
+<script>
+<!--Formset settings-->
+$j('.furnishing-formset').formset({
+    prefix: 'furnishings',
+    formTemplate: '#id_empty_form',
+    addText: 'add furnishing',
+    deleteText: 'remove',
+    addCssClass: 'furnishing-add',
+    deleteCssClass: 'furnishing-delete',
+    formCssClass: 'furnishing-dynamic-form',
+});
+
+<!--Get choices for a furnishing via POST-->
+function getChoices(furnishing, callback){
+    var data = {furnishing: furnishing, csrfmiddlewaretoken: csrf_token()};
+    $j.post('/manage/{{ prog.url }}/ajaxfurnishingchoices', data, "json").success(callback);
+}
+
+<!--Update the option field to reflect whether there are specific choices (dropdown) or not (open response). If a value is supplied, set that value as selected afterwards.-->
+function update_choices(obj, value) {
+    var furnishing = $j(obj).val()
+    getChoices(furnishing, function(response) {
+        var choices = response.choices;
+        var num = $j(obj).attr('id').split("-")[1];
```

Comment:
Yep, it will always be an integer that represents the index of the formset, thanks to the dynamic formset plugin (the ID of the furnishing field is always of the form `id_furnishings-[index]-furnishing`, while the ID of the choice field is always of the form `furnishings-[index]-choice`). Please do let me know if you know of a better way, though!